qiskit_ionq/ionq_client.py

```diff
@@ -77,6 +77,32 @@ def make_path(self, *parts):
         """
         return "/".join([self._url] + list(parts))
 
+    def _get_with_retry(self, req_path, params=None, headers=None, timeout=30):
+        """Make a GET request with retry logic and exception handling.
+
+        Args:
+            req_path (str): The URL path to make the request to.
+            params (dict, optional): Parameters to include in the request body.
+            headers (dict, optional): Headers to include in the request.
+            timeout (int, optional): Timeout for the request.
+            query_params (dict, optional): Query parameters to include in the URL.
+
+        Raises:
+            IonQRetriableError: When a retriable error occurs during the request.
+
+        Returns:
+            Response: A requests.Response object.
+        """
+        try:
+            res = requests.get(
+                req_path, params=params, headers=headers, timeout=timeout
+            )
+            res.raise_for_status()
+        except requests.exceptions.RequestException as req_exc:
```

Comment:
if you remove raise_for_status you can just catch any error here, as they should all relate to the socket i/o